packages/tailwind/src/tailwind.tsx

```diff
@@ -1,179 +1,142 @@
+  if (child.type === "head" || (child as any).type.displayName === "Head") {
+    const styleElement = <style>{responsiveStyles}</style>;
+
+    const headChildren = React.Children.toArray(child.props.children);
+    headChildren.push(styleElement);
 
-/**
- * Clean css selectors to replace all non-alphanumeric characters with underscores
- */
-function cleanCss(css: string) {
-  let newCss = css
-    .replace(/\\/g, "")
-    // find all css selectors and look ahead for opening and closing curly braces
-    .replace(/[.\!\#\w\d\\:\-\[\]\/\.%\(\))]+(?=\s*?{[^{]*?\})\s*?{/g, (m) => {
-      return m.replace(/(?<=.)[:#\!\-[\\\]\/\.%]+/g, "_");
-    })
-    .replace(/font-family(?<value>[^;\r\n]+)/g, (m, value) => {
-      return `font-family${value.replace(/['"]+/g, "")}`;
+    child = React.cloneElement(child, child.props, ...headChildren);
+  }
+  if (child.props.children) {
+    const children = React.Children.toArray(child.props.children);
+    const processedChildren = children.map((processedChild) => {
+      if (React.isValidElement(processedChild)) {
+        return processHead(processedChild, responsiveStyles);
+      }
+      return child;
     });
-  return newCss;
-}
 
-/**
- * Get media query css to put in head
- */
-function getMediaQueryCss(css: string) {
-  const mediaQueryRegex = /@media[^{]+\{(?<content>[\s\S]+?)\}\s*\}/gm;
-
-  return (
-    css
-      .replace(mediaQueryRegex, (m) => {
-        return m.replace(
-          /([^{]+\{)([\s\S]+?)(\}\s*\})/gm,
-          (_, start, content, end) => {
-            const newContent = (content as string).replace(
-              /(?:[\s\r\n]*)?(?<prop>[\w-]+)\s*:\s*(?<value>[^};\r\n]+)/gm,
-              (_, prop, value) => {
-                return `${prop}: ${value} !important;`;
-              },
-            );
-            return `${start}${newContent}${end}`;
-          },
-        );
-      })
-      // only return media queries
-      .match(/@media\s*([^{]+)\{([^{}]*\{[^{}]*\})*[^{}]*\}/g)
-      ?.join("") ?? ""
-  );
+    child = React.cloneElement(child, child.props, ...processedChildren);
+  }
+
+  return child;
 }
 
-/**
- * Make a map of all class names and their css styles
- */
-function makeCssMap(css: string) {
-  const cssNoMedia = css.replace(
-    /@media[^{]+\{(?<content>[\s\S]+?)\}\s*\}/gm,
-    "",
+export const Tailwind: React.FC<TailwindProps> = ({ children, config }) => {
+  const headStyles: string[] = [];
+
+  const childrenWithInlineStyles = React.Children.map(children, (child) => {
+    if (React.isValidElement(child)) {
+      return processElement(child, headStyles, config);
+    }
+    return child;
+  });
+
+  if (!childrenWithInlineStyles) return <>{children}</>;
+
+  const fullHTML = renderToStaticMarkup(<>{childrenWithInlineStyles}</>);
+
+  const hasResponsiveStyles = /@media[^{]+\{(?<content>[\s\S]+?)\}\s*\}/gm.test(
+    headStyles.join(" "),
   );
+  const hasHTML = /<html[^>]*>/gm.test(fullHTML);
```

Comment:
can this and `hasHead` be done in one pass so we don't test the full html twice?